e2e/volume_ops_test.go

```diff
@@ -148,8 +151,51 @@ func testVolumeStop(t *testing.T) {
 
 func testVolumeList(t *testing.T) {
 	r := require.New(t)
+	filterParams := map[string]string{
+		"key":   "owner",
+		"value": "gd2test",
+	}
+	volumes, err := client.Volumes("", filterParams)
+	r.Nil(err)
+	r.Len(volumes, 1)
```

Comment:
When both key and value is present, return only those volumes where both matches. This doesn't seem to do that as per the code above in `testVolumeCreate` where you've given only key.